.github/workflows/nightly_detect_failure.yml

```
@@ -44,7 +44,14 @@ jobs:
         toolchain=$(<lean-toolchain)
         if [[ $toolchain =~ leanprover/lean4:nightly-([a-zA-Z0-9_-]+) ]]; then
           version=${BASH_REMATCH[1]}
-          git push origin refs/heads/nightly-testing:refs/heads/nightly-testing-$version
+          if git ls-remote --tags origin refs/tags/nightly-testing-$version | grep -q 'refs/tags/nightly-testing-$version'; then
```

Possible alternative
```suggestion
          if git ls-remote --exit-code origin "refs/tags/nightly-testing-$version" >/dev/null; then
```